bundler/lib/bundler/index.rb

```diff
@@ -170,31 +160,41 @@ def add_source(index)
 
     private
 
+    def search_by_query(query)
```

Comment:
There were no callers except the one here. Moved to private and changed to return nil when nothing found so calling code can more easily reduce array usage.